src/app/coins.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { Coins } from './coins'
+import { HttpClient } from '@angular/common/http';
+import { catchError, map, tap } from 'rxjs/operators';
+
+
+@Injectable()
+export class CoinsService {
+
+	private coinsUrl: string = 'http://coincap.io/front';
+	public lastFetchDateStamp: Date;
+
+	constructor(
+		private http: HttpClient
+	) { }
+
+	getCoins(): Observable<Coins[]> {
+		return this.http.get<Coins[]>(this.coinsUrl)
+			.pipe(
+				tap(coins => {
+					this.lastFetchDateStamp = new Date();
```

Comment:
To równie dobrze możesz uzyskać w komponencie, chyba nie ma sensu wystawianie tego w serwisie i potem dopiero konsumowanie tego w komponencie.